### PR TITLE
feat: adds year to WiO tweet threads

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

Add year to WiO

## Fixes

-
## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [x] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

-

### CODE REVIEW

File: packages/valory/skills/olas_week_abci/prompts.py [hunk 1]
```diff
@@ -33,7 +33,7 @@
 and so on
 
 Your response should follow the format:
-"Week [Week number] in Olas
+"Week [Week number] '23 in Olas
```
Comment (DavidMinarsch) [inline packages/valory/skills/olas_week_abci/prompts.py:36]: Turns out these lines of the template and the matching line of the response are not actually used. So can just hardcode like this.

File: packages/valory/skills/olas_week_abci/behaviours.py [hunk 2]
```diff
@@ -130,7 +130,7 @@ def build_thread(raw_text: str, week: int) -> list:
     ]
 
     # Build tweets. Add highlights while the max char count is not exceeded
-    header = f"Week {week} in Olas\n\nHighlights included:\n"
+    header = f"Week {week} '{year} in Olas\n\nHighlights included:\n"
```
Comment (DavidMinarsch) [inline packages/valory/skills/olas_week_abci/behaviours.py:133]: This is the relevant change

File: packages/valory/skills/olas_week_abci/behaviours.py [hunk 3]
```diff
@@ -710,10 +711,11 @@ def evaluate_summary(
         llm_dialogues = cast(LlmDialogues, self.context.llm_dialogues)
 
         # llm request message
+        year_abbreviation = int(str(year)[2:])
         request_llm_message, llm_dialogue = llm_dialogues.create(
             counterparty=str(LLM_CONNECTION_PUBLIC_ID),
             performative=LlmMessage.Performative.REQUEST,
-            prompt_template=tweet_summarizer_prompt.replace("{tweet_text}", text),
+            prompt_template=tweet_summarizer_prompt.format(tweet_text=text),
```
Comment (dvilelaf) [inline packages/valory/skills/olas_week_abci/behaviours.py:718]: I think there was an issue with using format here, but can't recall it what was it. Let's give it a try with format again.

File: packages/valory/skills/olas_week_abci/prompts.py (hunk 1 above)
Comment (dvilelaf) [reply packages/valory/skills/olas_week_abci/prompts.py:36]: Shouldnt we pass the year here as a variable as well? Or at least use 24

File: packages/valory/skills/olas_week_abci/prompts.py [hunk 4]
```diff
@@ -114,7 +114,7 @@
 https://twitter.com/autonolas/status/1710294338435957248"
 
 Your response should have been:
-"Week 40 in Olas
+"Week 40 '23 in Olas
```
Comment (dvilelaf) [inline packages/valory/skills/olas_week_abci/prompts.py:117]: 24

Comment (Adamantios) [reply packages/valory/skills/olas_week_abci/prompts.py:117]: Same question as in https://github.com/valory-xyz/IEKit/pull/117#discussion_r1442853787.

File: packages/valory/skills/olas_week_abci/behaviours.py (hunk 3 above)
Comment (Adamantios) [reply packages/valory/skills/olas_week_abci/behaviours.py:718]: Importing `tweet_summarizer_prompt` and running `tweet_summarizer_prompt.format(tweet_text="test")` works locally.

**Review by angrybayblade — APPROVED** [review]

File: packages/valory/skills/olas_week_abci/prompts.py (hunk 1 above)
Comment (dvilelaf) [reply packages/valory/skills/olas_week_abci/prompts.py:36]: Yes, that's right.

**Pull request merged**